blas/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

```diff
@@ -501,4 +501,118 @@ KOKKOSBLAS3_CGEMM_ROCBLAS(Kokkos::LayoutRight, Kokkos::HIPSpace, false)
 }  // namespace KokkosBlas
 #endif  // KOKKOSKERNELS_ENABLE_TPL_ROCBLAS
 
+
+#if defined(KOKKOSKERNELS_ENABLE_TPL_MKL) && defined(KOKKOS_ENABLE_SYCL)
+#include <KokkosBlas_tpl_spec.hpp>
+#include <oneapi/mkl/blas.hpp>
+
+namespace KokkosBlas::Impl {
+
+#define KOKKOSBLAS3_XGEMM_MKL(SCALAR_TYPE,          \
+                                  LAYOUT, MEM_SPACE,             \
+                                  ETI_SPEC_AVAIL)                            \
+  template <class ExecSpace>                                                 \
+  struct GEMM<ExecSpace,                                                     \
+              Kokkos::View<const SCALAR_TYPE**, LAYOUT,                      \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              Kokkos::View<const SCALAR_TYPE**, LAYOUT,                      \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              Kokkos::View<SCALAR_TYPE**, LAYOUT,                            \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              true, ETI_SPEC_AVAIL> {                                        \
```

Comment:
Can you replace `ETI_SPEC_AVAIL` with:
```
gemm_tpl_spec_avail<ExecSpace, 
                                        Kokkos::View<const SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE, Kokkos::MemoryTraits<Kokkos::Unmanaged>>,
                                        Kokkos::View<const SCALAR_TYPE**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, Kokkos::MemoryTraits<Kokkos::Unmanaged>>,
                                        Kokkos::View<SCALAR_TYPE**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, Kokkos::MemoryTraits<Kokkos::Unmanaged>>::value>
```
With that you can remove the ETI_SPEC_AVAIL from the list of macro parameters.

Reply:
`norm1` is like that, but I think that means this TPL will only be available if the type combo was ETI'd as well.

Reply:
That's correct I think my snippet above is wrong, my bad for typing it too fast.
It should really read `gemm_eti_spec_avail<...>` which means we actually query the struct that knows if this eti is available or not instead of blindly trying both which ends up instantiating twice the amount of kernels...